### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by AlmazYarullin**

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  end
+
+  def create
+    authorize Comment, :create?
+    comment = Comment.new(comment_params)
+
+    if comment.save
+      redirect_to tasks_path, notice: 'Comment was successfully created.'
+    else
+      redirect_to comment.task, alert: 'Comment was not created'
```
Comment (DmitryBarskov) [inline app/controllers/comments_controller.rb]: How does user know what exactly is wrong with the comment?

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  end
+
+  def edit
+  end
+
+  def update
+    if @comment.update(comment_params)
+      redirect_to @comment.task, notice: 'Comment was successfully updated.'
+    else
+      redirect_to @comment.task, alert: 'Comment was not updated'
```
Comment (DmitryBarskov) [inline app/controllers/comments_controller.rb]: Same thing

File: app/models/user.rb [hunk 3]
```diff
@@ -1,7 +1,11 @@
...
   has_secure_password
 
-  has_many :projects
+  has_and_belongs_to_many :projects
 
   validates :email, presence: true, uniqueness: true
+
+  def to_s
+    email
+  end
```
Comment (DmitryBarskov) [inline app/models/user.rb:10]: What for?

File: app/policies/comment_policy.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+class CommentPolicy < ApplicationPolicy
+  def create?
+    user.present?
```
Comment (DmitryBarskov) [inline app/policies/comment_policy.rb]: Maybe user has to have access to a task?

File: app/policies/comment_policy.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  end
+
+  def edit?
+    user.present?
+  end
+
+  def update?
+    user.present?
```
Comment (DmitryBarskov) [inline app/policies/comment_policy.rb]: Maybe user has to be the author of the task?

File: app/views/comments/edit.html.slim [hunk 6]
```diff
@@ -0,0 +1,3 @@
+h1 Editing comment
+
+= render 'form', comment: @comment
```
Comment (DmitryBarskov) [inline app/views/comments/edit.html.slim]: This won't work since form url is wrong

File: db/migrate/20211023080328_add_user_to_project.rb [hunk 7]
```diff
@@ -1,5 +1,5 @@
 class AddUserToProject < ActiveRecord::Migration[6.1]
-  def down
+  def up
```
Comment (DmitryBarskov) [inline db/migrate/20211023080328_add_user_to_project.rb]: change

**Review by DmitryBarskov — CHANGES_REQUESTED** [review]

File: app/views/comments/edit.html.slim (hunk 6 above)
Comment (Pumpka-s) [reply app/views/comments/edit.html.slim:3]: You mean it should be like `= render 'comments/form', comment: @comment`?

File: db/migrate/20211023080328_add_user_to_project.rb (hunk 7 above)
Comment (AlmazYarullin) [reply db/migrate/20211023080328_add_user_to_project.rb]: Oops, my bad

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (AlmazYarullin) [reply app/controllers/comments_controller.rb]: Will be fixed

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (AlmazYarullin) [reply app/controllers/comments_controller.rb]: Same answer

File: app/policies/comment_policy.rb (hunk 4 above)
Comment (AlmazYarullin) [reply app/policies/comment_policy.rb]: I forgot to change the policy... I'll fix it next commit

File: app/policies/comment_policy.rb (hunk 5 above)
Comment (AlmazYarullin) [reply app/policies/comment_policy.rb]: Sure

File: app/views/comments/edit.html.slim (hunk 6 above)
Comment (AlmazYarullin) [reply app/views/comments/edit.html.slim]: Fixed

File: app/models/user.rb (hunk 3 above)
Comment (AlmazYarullin) [reply app/models/user.rb:10]: It will come in handy next homework

**Review by DmitryBarskov — APPROVED** [review]

**Review by DmitryBarskov — APPROVED** [review]

**Review by DmitryBarskov — APPROVED** [review]

**Pull request merged**